tizzo/lepew
# Feature/update docker

This PR aims to achieve the dream of having a single command that, once run, will leave a docker image that can be used directly in probo builds.

### Notable changes
 -  Updates the puppet modules used for provisioning.
 - Removes fetcher from the container.
 - Disables opcache in the container.
 - Remove `/var/www/html` and `/var/www/default` folders
 - Creates a default vhost that is enabled as zzz-default.conf and points to `/var/www/html`
 - Sets drush presets in `/etc/drush/conf.d/ 
    - root defaults to /var/www/html
    - `yes` flag is set to to true
 - Install ruby-dev and ruby-compass
 - Installs gem bundler
 - Installs drupal console
 - Install wp-cli
 - Upgrades drush to 8.0.3 (current latest)
 - Installs mcrypt
 - Installs image-magic

### To Test
 - `vagrant up` should build a VM, provision it and then build a docker image in that VM.
 - Log in to the VM via vagrant ssh
 - sudo su
 - `docker images` to list the available images. You should see a new one with no name.
 - `docker run -it [SOME-HASH] bash`
 - Check bundler: `bundler --version`
 - Check drush: `drush --version`
 - `ls /var/www/` should show no folders
 - `ls /etc/apache2/sites-available` should show a default.conf file.
 - `ls /etc/apache2/sites-enabled` should show a zzz-default.conf file which is a symlink to the previously mentioned file.
 - `wp --version --allow-root` to ensure the latest wp-cli
- use phpinfo() via a cli script to check that imagemagick version is 3.1.2
- `which compass` reveals `/usr/bin/compass` (compass version doesn't work for me)
- check the php.ini for apache and ensure that opcache is not enabled.
